src/hardware/mixer.cpp

```diff
@@ -326,10 +336,13 @@ static ReverbPreset reverb_pref_to_preset(const std::string_view pref)
 	return ReverbPreset::None;
 }
 
-static void configure_reverb(std::string reverb_pref)
+static void configure_reverb(const std::string_view reverb_pref)
 {
 	auto was_reverb_on = mixer.do_reverb;
-	mixer.do_reverb = (reverb_pref != "off");
+
+	const auto reverb_pref_has_bool = parse_bool_setting(reverb_pref);
+
+	mixer.do_reverb = (!reverb_pref_has_bool || *reverb_pref_has_bool == true);
```

Comment:
I think this criteria is wrong? Should be:

```cpp
mixer.do_reverb = (reverb_pref_has_bool && *reverb_pref_has_bool == false) ? false : true;
``` 

Because reverb can be `off`, `disabled`, `no` etc. for false, but if it `on`, `enabled`, etc. _or_ if it's set to a preset name like `small`, `medium`, etc, then it should be enabled.

Reply:
Better yet, `reverb_pref_to_preset` handles this already. So maybe call that, and do:

```
mixer.do_reverb = (reverb_pref_to_preset(reverb_pref) != ReverbPreset::None);
```

Reply:
> I think this criteria is wrong?

`mixer.do_reverb = (!reverb_pref_has_bool || *reverb_pref_has_bool == true);`

- The first criteria (`!reverb_pref_has_bool`) checks if a custom value is provided (so I think that's OK).

- The second criteria only gets tested if the first isn't true. So if we're not using a non-custom value (in other words, if a bool was provided), then test if the bool was positive (so I think that's OK, too).

(given the <optional> itself is testable as boolean combined with the content itself being bool, I find it a bit hard on the eyes!)

Definitely like the re-use of `reverb_pref_to_preset` better.